### ISSUE TITLE
Parser fails when used in a ESM context

### ISSUE THREAD

**Issue by gtm-nayan** [TyrealHu/acorn-typescript#6]
It fails to parse something like `1 as number` because the plugin does reference equality checks on token types.

But if using it from an ESM file, the users will load acorn/index.mjs, whereas this plugin will load the tokenTypes from acorn/index.js, which means the token types will be different objects even if they are the same token type and the reference equality will return false.

Possible solutions:
1. Replace tokType reference equality checks with comparing tokType.label
2. Rename the `index.esm.js` file to `index.mjs` and let users import from it manually
3. Use exports map instead of module since node doesn't care about module and will always load the CJS version of this package (https://publint.dev/acorn-typescript@1.2.5)

**Comment by TyrealHu** [comment]
Could u show me the case which fails to parse something like 1 as number

**Comment by gtm-nayan** [comment]
```js
// index.js
import * as acorn from 'acorn';
import { tsPlugin } from 'acorn-typescript';

const source = `1 as number`;
const node = acorn.Parser.extend(tsPlugin()).parse(source, {
	sourceType: 'module',
	ecmaVersion: 'latest',
	locations: true
});

console.log(node);
```

package.json
```json
{"type": "module"}
```

will fail with `SyntaxError: Unexpected token (1:5)`

**Comment by TyrealHu** [comment]
This issue has been fixed in version [1.2.7](https://www.npmjs.com/package/acorn-typescript/v/1.2.7)

**Comment by gtm-nayan** [comment]
Thanks for the quick fix.
